### PR TITLE
feat: Add non google default creds provider to subscribers in Spark Connector

### PR DESCRIPTION
**Pull request by jiangmichaellll**
As titled.

### CODE REVIEW

File: pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousInputPartition.java [hunk 1]
```diff
@@ -33,14 +32,17 @@
 public class PslContinuousInputPartition
     implements ContinuousInputPartition<InternalRow>, Serializable {
 
+  private final SubscriberFactory subscriberFactory;
```
Comment (dpcollins-google) [inline pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousInputPartition.java:35]: is this serializable?

File: pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousReader.java [hunk 2]
```diff
@@ -108,6 +113,14 @@ public StructType readSchema() {
         .map(
             v ->
                 new PslContinuousInputPartition(
+                    (consumer) ->
```
Comment (dpcollins-google) [inline pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousReader.java:116]: Can you push this up one more level? Have PslContinuousReader take a SubscriberFactory.

Comment (dpcollins-google) [reply pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousReader.java:116]: looks like all it needs from this class is the partition. So can you make an interface instead that has

Subscriber newSubscriber(Partition partition, Consumer<List<SequencedMessage>> messages)

and is serializable?

File: pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousInputPartition.java (hunk 1 above)
Comment (jiangmichaellll) [reply pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousInputPartition.java:35]: yes

File: pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousReader.java (hunk 2 above)
Comment (jiangmichaellll) [reply pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousReader.java]: done.

File: google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberBuilder.java [hunk 3]
```diff
@@ -51,6 +51,18 @@ public static Builder newBuilder() {
     return new AutoValue_SubscriberBuilder.Builder().setContext(PubsubContext.of());
   }
```
Comment (dpcollins-google) [inline google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberBuilder.java:53]: Put this in ServiceClients.java instead.

File: google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberBuilder.java [hunk 4]
```diff
@@ -77,19 +89,17 @@ public Subscriber build() throws ApiException {
...
-                  .putAll(metadata)
-                  .putAll(routingMetadata)
-                  .build();
+          SubscriberServiceSettings.Builder settingsBuilder =
+              SubscriberServiceSettings.newBuilder();
+          addDefaultMetadata(
+              autoBuilt.context(),
+              autoBuilt.subscriptionPath(),
+              autoBuilt.partition(),
+              SubscriberServiceSettings.newBuilder());
```
Comment (dpcollins-google) [inline google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberBuilder.java:98]: This should be settingsBuilder: As of now, it modifies an object then discards it.

File: google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberBuilder.java [hunk 5]
```diff
@@ -51,6 +51,18 @@ public static Builder newBuilder() {
     return new AutoValue_SubscriberBuilder.Builder().setContext(PubsubContext.of());
   }
 
+  public static void addDefaultMetadata(
+      PubsubContext context,
```
Comment (dpcollins-google) [inline google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberBuilder.java:55]: instead, have this take:

PubsubContext, RoutingMetadata, Builder

where Builder is declared like in addDefaultMetadata.

**Review by dpcollins-google — CHANGES_REQUESTED** [review]
Fix changes then LGTM

**Review by dpcollins-google — APPROVED** [review]

**Pull request merged**